src/Scanner.js

```diff
@@ -134,7 +134,13 @@ class DefaultScanner {
         let element = input
         // traverse the DOM to search for related inputs
         while (element.parentElement && element.parentElement !== document.body) {
+            // If parent includes a form return the current element to avoid overlapping forms
+            if (element.parentElement?.querySelector('form')) {
+                return element
+            }
```

Comment:
@GioSensation ℹ️ The below is mostly brain bump of thoughts -- it may be inaccurate and certainly non-blocking.

I wonder if we could defend against this issue in our `*_SELECTOR` values below so that they always ignore inputs which are nested within a form.

I don't have a real world example, but for the contrived example below, would autofill now not detect the right parent element?
The algorithm would now traverse up from email input, but at `div#email` would detect the wrapping parent `div` contains a `form`, and so would return `div#email`. Does this matter?
```html
<form>
    <div>
        <div id="email">
            <label>Email</label>
            <input name="email" />
        </div>
        <form>
            <button>Subscribe to newsletter</button>
        </form>
    </div>
    <div id="password">
        <label>Password</label>
        <input name="password" />
    </div>
    <button>Submit</button>
</form>
```

Reply:
It doesn't matter. In this case both inputs already are part of a form so we don't traverse the DOM at all.